libtelemetry/src/main/java/com/mapbox/android/telemetry/CertificateBlacklist.java

```diff
@@ -0,0 +1,226 @@
+  private static final String RETRIEVE_TIME_FAIL = "Unable to retrieve last update time from blacklist";
+  private static final String RETRIEVE_BLACKLIST_FAIL = "Unable to retrieve blacklist contents from file";
+  private static final String REQUEST_FAIL = "Request failed to download blacklist";
+  private static final String READLINE_FAIL = "Unable to read line of Blacklist file";
+  private static final String HTTPS_SCHEME = "https";
+  private static final Map<Environment, String> ENDPOINTS = new HashMap<Environment, String>() {
+    {
+      put(Environment.COM, COM_CONFIG_ENDPOINT);
+      put(Environment.STAGING, COM_CONFIG_ENDPOINT);
+      put(Environment.CHINA, CHINA_CONFIG_ENDPOINT);
+    }
+  };
+  private final Context context;
+  private final String accessToken;
+  private Logger logger;
+
+  CertificateBlacklist(Context context, String accessToken) {
+    this.context = context;
+    this.accessToken = accessToken;
+    this.logger = new Logger();
+  }
+
+  List<String> retrieveBlackList() {
+    File directory = context.getFilesDir();
+    List<String> blacklist = new ArrayList<>();
+
+    if (directory.isDirectory()) {
+      File file = new File(directory, BLACKLIST_FILE);
+
+      if (file.exists()) {
+        try {
+          blacklist = obtainBlacklistContents(file);
+          blacklist.remove(BLACKLIST_HEAD);
+        } catch (IOException exception) {
+          logger.error(RETRIEVE_BLACKLIST_FAIL, exception.getMessage());
+        }
+      }
+    }
+
+    return blacklist;
+  }
+
+  boolean daySinceLastUpdate() {
+    long millisecondDiff = System.currentTimeMillis() - retrieveLastUpdateTime();
+    return millisecondDiff >= DAY_IN_MILLIS;
+  }
+
+  private long retrieveLastUpdateTime() {
+    File directory = context.getFilesDir();
+    File file = new File(directory, BLACKLIST_FILE);
+
+    long lastUpdateTime = 0;
+
+    if (file.exists()) {
+      try {
+        List<String> blacklist = obtainBlacklistContents(file);
+        lastUpdateTime = Long.valueOf(blacklist.get(BLACKLIST_HEAD));
+      } catch (IOException exception) {
+        logger.error(RETRIEVE_TIME_FAIL, exception.getMessage());
+      }
+    }
+
+    return lastUpdateTime;
+  }
+
+  void updateBlacklist() {
+    HttpUrl requestUrl = new HttpUrl.Builder().scheme(HTTPS_SCHEME)
+      .host(determineConfigEndpoint())
+      .addPathSegment("events-config")
+      .addQueryParameter(ACCESS_TOKEN_QUERY_PARAMETER, accessToken)
+      .build();
+
+    Request request = new Request.Builder()
+      .url(requestUrl)
+      .build();
+
+    OkHttpClient client = new OkHttpClient();
+    client.newCall(request).enqueue(this);
+  }
+
+  private void saveBlackList(List<String> revokedKeys) {
+    String fileContents = createListContent(revokedKeys);
+    FileOutputStream outputStream;
+
+    try {
+      outputStream = context.openFileOutput(BLACKLIST_FILE, Context.MODE_PRIVATE);
+      outputStream.write(fileContents.getBytes());
+      outputStream.close();
```

Comment:
what if write() on line 134 throws? shouldn't close() be called in finally block?